### PR TITLE
Single field interface

### LINKED ISSUE #80: can't get grib_util to build with ip-4.0.0
I have been trying to get grib_util to build with ip-4.0.0, but no luck.

In the CMakeLists.txt I added:

```
# For NCEPLIBS-ip-4.0.0 and later, we need to add a "use" statement to
# the code. Check for this...
if(ip_VERSION GREATER_EQUAL 4.0.0)
  add_compile_definitions(USE_IP_MOD)
  message(STATUS "ip library > 4.0.0, USE_IP_MOD set")
else()
  message(STATUS "ip library < 4.0.0, USE_IP_MOD not set")
endif()
```

And this seems to work, when compiling with ip-4.0.0 I see:

`cd /home/ed/NCEPLIBS-grib_util/b/src/copygb && /usr/bin/gfortran -DUSE_IP_MOD -I/usr/local/NCEPLIBS-bacio-2.5.0/include -I/usr/local/NCEPLIBS-w3emc-2.9.2/include_d -I/usr/local/NCEPLIBS-ip-4.0.0/include_d -I/usr/local/NCEPLIBS-sp-2.3.3/include_d -g -fdefault-real-8  -O3 -g -fdefault-real-8  -c /home/ed/NCEPLIBS-grib_util/src/copygb/copygb.F -o CMakeFiles/copygb.dir/copygb.F.o
`

So the USE_IP_MOD pre-processor define is being properly set.

In the code I have:

```
      PROGRAM COPYGB
#ifdef USE_IP_MOD
      use ip_mod
#endif

```

But when I try to build:

```
cd /home/ed/NCEPLIBS-grib_util/b/src/copygb && /snap/cmake/1000/bin/cmake -E cmake_link_script CMakeFiles/copygb.dir/link.txt --verbose=1
/usr/bin/gfortran -g -fdefault-real-8  -O3 -g -fdefault-real-8  CMakeFiles/copygb.dir/copygb.F.o -o copygb  /usr/local/NCEPLIBS-bacio-2.5.0/lib/libbacio.a /usr/local/NCEPLIBS-w3emc-2.9.2/lib/libw3emc_d.a /usr/local/NCEPLIBS-ip-4.0.0/lib/libip_d.a /usr/local/NCEPLIBS-sp-2.3.3/lib/libsp_d.a /usr/local/NCEPLIBS-bacio-2.5.0/lib/libbacio.a 
/usr/bin/ld: CMakeFiles/copygb.dir/copygb.F.o: in function `intgrib1_':
/home/ed/NCEPLIBS-grib_util/src/copygb/copygb.F:1225: undefined reference to `ipolates_'
/usr/bin/ld: /home/ed/NCEPLIBS-grib_util/src/copygb/copygb.F:1300: undefined reference to `ipolatev_'
/usr/bin/ld: /home/ed/NCEPLIBS-grib_util/src/copygb/copygb.F:1331: undefined reference to `ipolatev_'
/usr/bin/ld: /home/ed/NCEPLIBS-grib_util/src/copygb/copygb.F:1290: undefined reference to `ipolatev_'
/usr/bin/ld: /home/ed/NCEPLIBS-grib_util/src/copygb/copygb.F:1231: undefined reference to `ipolates_'
/usr/bin/ld: /home/ed/NCEPLIBS-grib_util/src/copygb/copygb.F:1268: undefined reference to `ipolatev_'
/usr/bin/ld: /home/ed/NCEPLIBS-grib_util/src/copygb/copygb.F:1253: undefined reference to `ipolates_'
/usr/bin/ld: /home/ed/NCEPLIBS-grib_util/src/copygb/copygb.F:1212: undefined reference to `ipolates_'
/usr/bin/ld: CMakeFiles/copygb.dir/copygb.F.o: in function `MAIN__':
/home/ed/NCEPLIBS-grib_util/src/copygb/copygb.F:260: undefined reference to `makgds_'
collect2: error: ld returned 1 exit status
make[2]: *** [src/copygb/CMakeFiles/copygb.dir/build.make:101: src/copygb/copygb] Error 1
make[2]: Leaving directory '/home/ed/NCEPLIBS-grib_util/b'
make[1]: *** [CMakeFiles/Makefile2:1075: src/copygb/CMakeFiles/copygb.dir/all] Error 2
make[1]: Leaving directory '/home/ed/NCEPLIBS-grib_util/b'
make: *** [Makefile:146: all] Error 2

```

What am I doing wrong?

There is a branch in grib_util called ejh_ip which has all these changes...

### PR DESCRIPTION
**Pull request by kgerheiser**
Add an interface for grib1 ipolates/ipolatev when `km` (number of fields to interpolate) equals 1. The interface removes the `km` dimension from the input arrays. This allows backwards compatibility with how the library was used with no interface checking.

Add `makgds.f90` which was removed when combining ip/ip2.

With these changes I was able to build grib_util with just the addition of `use ip_mod`.

Fix #80

### CODE REVIEW

File: src/ipolates.f90 [hunk 1]
```diff
@@ -101,6 +102,85 @@ subroutine ipolates_grid(ip, ipopt, grid_in, grid_out, mi, mo, km,&
 
   end subroutine ipolates_grid
 
+  !> @brief Special case of ipolates_grib1 when interpolating a single field.
```
Comment (edwardhartnett) [inline src/ipolates.f90]: No need for breif tag on subroutines. The first sentence will automatically be used for the brief description.

File: src/ipolates.f90 [hunk 2]
```diff
@@ -101,6 +102,85 @@ subroutine ipolates_grid(ip, ipopt, grid_in, grid_out, mi, mo, km,&
...
+  !! @param[out] lo  Output bitmaps (always output).
+  !! @param[out] go  Output fields interpolated.
+  !! @param[out] iret Return code.
+  !! - 0 Successful interpolation.
+  !! - 1 Unrecognized interpolation method.
+  !! - 2 Unrecognized input grid or no grid overlap.
+  !! - 3 Unrecognized output grid.
+  !! - 1x Invalid bicubic method parameters.
+  !! - 3x Invalid budget method parameters.
+  !! - 4x Invalid spectral method parameters.
```
Comment (edwardhartnett) [inline src/ipolates.f90:145]: Add date and author tags to subroutine please.

File: src/makgds.f90 [hunk 3]
```diff
@@ -0,0 +1,169 @@
+ SUBROUTINE MAKGDS(IOPT,KGDS,GDS,LENGDS,IRET)
+!$$$  SUBPROGRAM DOCUMENTATION BLOCK
```
Comment (edwardhartnett) [inline src/makgds.f90]: Can you convert this docblock to doxygen please?

**Comment by kgerheiser** [comment]
Actually, @edwardhartnett I think that `makgds.f90` better belongs in w3emc.

makgds doesn't do anything related to interpolation, it is not called by anything else in ip, and it calls all these functions from w3emc. This dependency isn't modeled in the build system, and it would add a w3emc dependency to ip.

```
! SUBPROGRAMS CALLED:
!   FI633        EXTRACT INFO FROM GRIB-GDS
!   R63W72       CONVERT W3FI63 PARMS TO W3FI72 PARMS
!   W3FI71       MAKE ARRAY USED BY GRIB PACKER FOR GDS
!   W3FI74       CONSTRUCT GRID DEFINITION SECTION (GDS)
```

**Comment by edwardhartnett** [comment]
OK, can you move it there please?

**Comment by kgerheiser** [comment]
Done

**Comment by kgerheiser** [comment]
Could I get a review @edwardhartnett?

**Review by edwardhartnett — APPROVED** [review]

**Pull request merged**